uaclient/serviceclient.py

```diff
@@ -45,7 +46,7 @@ def request_url(self, path, data=None, headers=None, method=None):
             headers = self.headers()
         if headers.get('content-type') == 'application/json' and data:
             data = json.dumps(data).encode('utf-8')
-        url = getattr(self.cfg, self.cfg_url_base_attr) + path
+        url = urljoin(getattr(self.cfg, self.cfg_url_base_attr), path)
```

Comment:
Using `urljoin` will stop us from being able to configure base URLs that aren't at the root of a domain:

```
>>> urljoin('http://base/path', '/uh/oh')
'http://base/uh/oh'
```

I think using `posixpath.join` should avoid it.

Reply:
though posixpath.join has another iteresting behavior in that if second param starts with leading forward slash, you only get the second value

>>> posixpath.join('https://asdf/', '/oops')
'/oops'

Reply:
will sort it to use posixpath appropriately